pipeline_dp/sampling_utils.py

```diff
@@ -26,3 +27,25 @@ def choose_from_list_without_replacement(a: list, size: int) -> list:
     sampled_indices = np.random.choice(np.arange(len(a)), size, replace=False)
 
     return [a[i] for i in sampled_indices]
+
+
+def _compute_64bit_hash(v) -> int:
+    m = hashlib.sha1()
+    m.update(repr(v).encode())
+    return int(m.hexdigest()[:16], 16)
+
+
+class ValueSampler:
+    """Deterministic value sampler.
+
+    For a value it returns whether this value should be kept. The keeping
+    decision is deterministic for a fixed value. On average the decision to keep
+    happens in sampling_rate cases.
```

Comment:
nit: It happens in `sampling_rate*num_values` cases right? Maybe we can say instead `happens with probability sampling_rate`?

Reply:
It's tricky to write about probability, since "The keeping decision is deterministic for a fixed value"

I've updated the text. PTAL